stl/inc/xstring

```diff
@@ -1671,7 +1684,7 @@ _NODISCARD constexpr bool operator==(
     return _Lhs._Equal(_Rhs);
 }
 
-
+#if !_HAS_CXX20

```

Comment:
Stylistically, it's odd to have these conditional sections:
```c++
#if !meow
// ...stuff...
#endif // !meow

#if meow
// ...other stuff...
#endif // meow
```
when we could instead have a single if/else conditional:
#if meow
// ...other stuff...
#else // ^^^ meow / !meow vvv
// ...stuff...
#endif // ^^^ !meow ^^^
```
Is there any reason we shouldn't follow the latter more conventional style here and for the blocks on 4569 - 4663?